CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/NotesViewTopComponent.java

```diff
@@ -0,0 +1,164 @@
+
+public class NotesViewTopComponent extends JavaFxTopComponent<NotesViewPane> implements GraphReportListener {
+    
+    private final NotesViewController notesViewController;
+    private final NotesViewPane notesViewPane;
+
+    /**
+     * Creates a new NotesViewTopComponent.
+     */
+    public NotesViewTopComponent() {
+        
+        setName(Bundle.CTL_NotesViewTopComponent());
+        setToolTipText(Bundle.HINT_NotesViewTopComponent());
+        
+        initComponents();
+        
+        notesViewController = new NotesViewController(this);
+        notesViewPane = new NotesViewPane(notesViewController);
+        
+        initContent();
+    }
+    
+    @Override
+    protected void handleNewGraph(final Graph graph) {
+
+        if (needsUpdate() && graph != null) {
+            notesViewPane.selectAllFilters();
+            notesViewPane.clearNotes();
+            notesViewPane.prepareNotesViewPane(notesViewController);
+        }
+    }
+
+    @Override
+    protected void handleGraphOpened(final Graph graph) {
+
+        if (needsUpdate() && graph != null) {
+            notesViewPane.selectAllFilters();
+            notesViewPane.clearNotes();
+            notesViewPane.prepareNotesViewPane(notesViewController);
+        }
+    }
+
+    @Override
+    protected void handleGraphClosed(final Graph graph) {
+
+        if (needsUpdate() && graph != null) {
+            notesViewPane.clearNotes();
+            notesViewPane.prepareNotesViewPane(notesViewController);
+            notesViewPane.closeEdit();
+        }
+    }
+
+    @Override
+    protected void handleComponentOpened() {
+        
+        GraphReportManager.addGraphReportListener(this);
+        Graph activeGraph = GraphManager.getDefault().getActiveGraph();
+        
+        if (activeGraph != null) {
+            notesViewPane.clearNotes();
+            notesViewPane.prepareNotesViewPane(notesViewController);
+            // Ensures plugin reports created while the Notes View is not "open" will appear when it is opened later.
+            notesViewPane.setGraphReport(notesViewController);
+        }
+    }
+
+    @Override
+    protected void handleComponentClosed() {
+        
+        GraphReportManager.removeGraphReportListener(this);
+        notesViewPane.clearNotes();
+        notesViewPane.prepareNotesViewPane(notesViewController);
+        notesViewPane.closeEdit();
+    }
+
+    @Override
+    protected String createStyle() {
+        return "resources/notes-view.css";
+    }
+
+    @Override
+    protected NotesViewPane createContent() {
+        return notesViewPane;
+    }
+
+    // Triggers when plugin reports are added or removed.
+    @Override
+    public void newPluginReport(PluginReport pluginReport) {
+        // Omit plugin reports from the Notes View and Quality Control View.
+        if (!pluginReport.getPluginName().contains("Notes View")) {
+            if (!pluginReport.getPluginName().contains("Quality Control View")) {
```

Comment:
merge these nested if statements

Reply:
This was a deliberate decision made in lieu of trying to merge the statements. If I merge these with a || it does not achieve the same effect of blocking plugin reports with either cases and causes an infinite loop which ends up locking up and crashing constellation.

Reply:
@sol695510 So you actually merge nested ifs with `&&` rather than `||` (with the outside if condition listed first). We know `&&` will work as both if conditions need to be true in order for the nested block code to run (the `||` will obviously then behave differently as then only one of these would  need to be true)

So what you're looking for here is:
`!pluginReport.getPluginName().contains("Notes View") && !pluginReport.getPluginName().contains("Quality Control View")`

Reply:
That wouldn't work either because a single plugin report's name can't have both "Notes View" and "Quality Control" in it, since I'm blocking plugin reports that either have the name containing "Notes View" so a plugin report from the Notes View or the name "Quality Control" so a plugin from the Quality Control action.

That makes it sound like it should be a || and should work with a || but for some reason it does not.

Reply:
@sol695510 Don't forget that there is a not (`!`) in front of each of those statements so that statement actually reads the "name doesn't contain 'Notes View' and name doesn't contain 'Quality Control View'" rather than "name doesn't contain 'Notes View' and 'Quality Control View'" so it's checking for each of those individually rather than both of them together.

An equivalent statement to the one I said is `!(pluginReport.getPluginName().contains("Notes View") || pluginReport.getPluginName().contains("Quality Control View"))` but it is preferred to propagate the `!`

If it still doesn't make sense, I'm happy to chat about this further but if it's of any use, sonar will flag the same thing.

Reply:
I kinda understood so I've made the changes. Took me some time because I was also trying to rectify some minor issues.